pkg/clientconn/client.go

```diff
@@ -365,7 +366,17 @@ func AuthenticatedHTTPTransport(endpoint string, server mtls.Subject, baseTransp
 
 // GRPCConnection establishes a gRPC connection to the given server, using the given connection options.
 func GRPCConnection(dialCtx context.Context, server mtls.Subject, endpoint string, clientConnOpts Options, dialOpts ...grpc.DialOption) (*grpc.ClientConn, error) {
-	allDialOpts := make([]grpc.DialOption, 0, len(dialOpts)+3)
+	allDialOpts := make([]grpc.DialOption, 0, len(dialOpts)+5)
+
+	retryOpts := []grpc_retry.CallOption{
+		grpc_retry.WithBackoff(grpc_retry.BackoffExponential(100 * time.Millisecond)),
+		grpc_retry.WithMax(3),
+	}
+
+	allDialOpts = append(allDialOpts,
+		grpc.WithStreamInterceptor(grpc_retry.StreamClientInterceptor(retryOpts...)),
```

Comment:
This change would affect the `scanner` and other components of our stack, and I'm a bit concerned if we would like a change like that in other services.

If we want retry logic for roxctl, then we should add it there. And I would suggest that we move these options to roxctl only. `roxctl/common/flags.go`

Reply:
You are right. I hoped this change will not affect them that much but it's better to limit the scope.